aioresponses/core.py

```diff
@@ -299,19 +299,39 @@ def add(self, url: 'Union[URL, str, Pattern]', method: str = hdrs.METH_GET,
         ))
 
     async def match(
-            self, method: str, url: URL, **kwargs: Dict
+        self, method: str, url: URL,
+        allow_redirects: bool = True, **kwargs: Dict
     ) -> Optional['ClientResponse']:
-        for i, matcher in enumerate(self._matches):
-            if matcher.match(method, url):
-                response = await matcher.build_response(url, **kwargs)
+        history = []
+        while True:
+            for i, matcher in enumerate(self._matches):
+                if matcher.match(method, url):
+                    response = await matcher.build_response(
+                        url, allow_redirects=allow_redirects, **kwargs
+                    )
+                    break
+            else:
+                if history:
+                    raise Exception('Redirect leads to an unmocked url')
```

Comment:
All unmatched requests should raise `ClientConnectionError`
`return None` should be enough

Reply:
I thought it would be nice to show to the user that they forgot to configure that ... if you think it's unnecessary I'll remove it

Reply:
I think `ClientConnectionError` with current message should be enough :)

Reply:
... and it will show user reason :)

Reply:
addressed in 319b8d9